web-extensions/chrome/engine/datamodel.js

```diff
@@ -280,6 +273,26 @@ BmcComic.prototype.iterSources = function(iterFunc) {
     }
 };
 
+/**
+ * This method return the matching source (comparison being performed on the
+ * reader) or `null` if non matching was found.
+ *
+ * @param {string} sourceReader - The reader of the source we want to get.
+ *
+ * @return {BmcComicSource}
+ *
+ */
+BmcComic.prototype.getSource = function(sourceReader) {
+    for (var i = 0; i < this._sources.length; ++i) {
```

Comment:
You could use a js Array method such as `const matches = this._sources.find(source => source.reader == sourceReader);`
(Then of course, checking whether you got null, an item, or an array of items - is that even possible here ?)

Reply:
Oh true! I'm still not up-to-date with js tools...